azure-mgmt-cdn/src/main/java/com/microsoft/azure/management/cdn/CdnEndpoint.java

```diff
@@ -0,0 +1,937 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.management.cdn;
+
+import com.microsoft.azure.management.apigeneration.Fluent;
+import com.microsoft.azure.management.cdn.implementation.EndpointInner;
+import com.microsoft.azure.management.resources.fluentcore.arm.CountryISOCode;
+import com.microsoft.azure.management.resources.fluentcore.arm.models.ExternalChildResource;
+import com.microsoft.azure.management.resources.fluentcore.model.Settable;
+import com.microsoft.azure.management.resources.fluentcore.model.Wrapper;
+
+import java.util.List;
+
+/**
+ * An immutable client-side representation of an Azure CDN endpoint.
+ */
+@Fluent
+public interface CdnEndpoint extends
+        ExternalChildResource<CdnEndpoint, CdnProfile>,
+        Wrapper<EndpointInner> {
+
+    /**
+     * Gets the origin host header.
```

Comment:
minor, but for simple "getters" we do not need a description -- `@return host header` is enough